tests/integration/test_definitions.py

```diff
@@ -384,7 +455,8 @@ def define_tests():
     all_tests = {}
     for test_dict in [
         basic_checks,
-        server_tests,
+        server_basic_tests,
```

Comment:
So rather than re-enumerating things here, what about some alternate patterns like just adding the contents of these sub-dicts to all_tests directly when creating them, or even just having this function be a generator that yields the key: value pairs and a little helper outside that consumes it.  It looks like the names of these test group dicts are used more for documentation than anything, suggesting they don't really need to be directly created at all.

Those are more just some style choices than anything though; my main concern with the design of this (and the above refactors would hit this too as it'd still largely be the same design) is that there's a lot of silent failures/bugs that this dictionary type structure is susceptible to: update and direct key assignment ('dict[key]=newvalue') will happily overwrite the values of any keys that are already there leading to incorrect tests being run if developers aren't careful to scan all of the test names in here and avoid them.  Some error checking here would be great to at least warn of that so such collisions can be more easily traced/fixed.

Reply:
I agree with you that this is following the same design structure as the testing structure. From what I can see you are right that the names of the test are mainly for documentation. See my comment above, I am not sure about the original reason for making it this way but in order to change the design of the testing structure we would need to refactor and redesign the framework

Reply:
Yeah, that's fair on the refactoring.   I'd still recommend some error checking to catch the silent failures that can happen here when joining all these dictionaries together